framework/src/org/checkerframework/framework/type/TypesIntoElements.java

```diff
@@ -244,7 +244,8 @@ private static void addUniqueTypeCompounds(Types types, Symbol sym, List<TypeCom
         List<Attribute.TypeCompound> res = List.nil();
 
         for (Attribute.TypeCompound tc : tcs) {
-            if (!TypeAnnotationUtils.isTypeCompoundContained(types, raw, tc)) {
+            // Don't insert an annotation if one with the same name already exists.
+            if (!TypeAnnotationUtils.isTypeCompoundContainedByName(types, raw, tc)) {
```

Comment:
I will not have time to look at this in the next few days. You should look at sym to see whether it has alternatives to appending lists of TypeCompound. Building up the right `res` should be easy.